homeassistant/components/advantage_air/climate.py

```diff
@@ -0,0 +1,283 @@
+    """Platform setup isn't required."""
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir climate platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirAC(instance, ac_key))
+        for zone_key in instance["coordinator"].data["aircons"][ac_key]["zones"]:
+            # Only add zone climate control when zone is in temperature control
+            if (
+                instance["coordinator"].data["aircons"][ac_key]["zones"][zone_key][
+                    "type"
+                ]
+                != 0
+            ):
+                entities.append(AdvantageAirZone(instance, ac_key, zone_key))
+    async_add_entities(entities)
+    return True
+
+
+class AdvantageAirClimateEntity(CoordinatorEntity, ClimateEntity):
+    """AdvantageAir Climate class."""
+
+    def __init__(self, instance):
+        """Initialize the base Advantage Air climate entity."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.device = instance["device"]
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported temperature step."""
+        return PRECISION_WHOLE
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature."""
+        return 32
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature."""
+        return 16
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return self.device
+
+
+class AdvantageAirAC(AdvantageAirClimateEntity):
+    """AdvantageAir AC unit."""
+
+    def __init__(self, instance, ac_key):
+        """Initialize the Advantage Air AC climate entity."""
+        super().__init__(instance)
+        self.ac_key = ac_key
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["name"]
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-climate'
+
+    @property
+    def target_temperature(self):
+        """Return the current target temperature."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["setTemp"]
+
+    @property
+    def hvac_mode(self):
+        """Return the current HVAC modes."""
+        if self.coordinator.data["aircons"][self.ac_key]["info"]["state"] == STATE_ON:
+            return ADVANTAGE_AIR_HVAC_MODES.get(
+                self.coordinator.data["aircons"][self.ac_key]["info"]["mode"],
+                self.coordinator.data["aircons"][self.ac_key]["info"]["mode"],
```

Comment:
```suggestion
            return ADVANTAGE_AIR_HVAC_MODES.get(
                self.coordinator.data["aircons"][self.ac_key]["info"]["mode"]
```
Why would you reference the same thing twice?

Reply:
Its a get statement, so if there is a match for that key, return that key, if not, return the key itself.
At the time I probably thought I was smart as this would allow the integration to expose new HVAC modes that I haven't "mapped".
Would you recommend I return None if its not a mapped HVAC mode?

Reply:
Check dev docs but I don't think undocumented modes are supported

Reply:
Indeed not https://developers.home-assistant.io/docs/core/entity/climate/#hvac-modes